src/game/interaction.c

```diff
@@ -158,7 +158,8 @@ u32 determine_interaction(struct MarioState *m, struct Object *o) {
 
     if (interaction == 0 && action & ACT_FLAG_ATTACKING) {
         u32 flags = (MARIO_PUNCHING | MARIO_KICKING | MARIO_TRIPPING);
-        if (m->flags & flags) {
+        if ((action == ACT_PUNCHING || action == ACT_MOVE_PUNCHING || action == ACT_JUMP_KICK) ||
```

Comment:
I'm not sure about this change because actions are supposed to asign the flags, rather than a punch always creating a hit for example.

Reply:
I'd agree but vanilla sm64 does this

Reply:
oh i see you're right, sorry about that https://github.com/n64decomp/sm64/blob/9921382a68bb0c865e5e45eb594d9c64db59b1af/src/game/interaction.c#L187